### PR TITLE
Set payid metrics protocol version and server agent

### PR DESCRIPTION
**Pull request by nhartner**
## High Level Overview of Change

Updates the metrics config to set the PayID protocol version and the server agent

### Context of Change

This will allow the server and protocol versions for people running the reference implementation to be tracked within the metrics platform

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Just updated the config settings

## Test Plan

Run PayID server and verify that the new metrics are set and published to the Xpring metrics infra.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
   },
   "dependencies": {
     "@hapi/boom": "^9.1.0",
-    "@payid-org/server-metrics": "^1.0.0",
+    "@payid-org/server-metrics": "github:payid-org/server-metrics#nh/server-and-protocol-version",
```
Comment (nhartner) [inline package.json:43]: FIXME: I need to cut a new release of server-metrics and fix this dependency before merging.

File: src/config.ts [hunk 2]
```diff
@@ -59,6 +59,12 @@ const config = {
     /** How frequently (in seconds) to refresh the PayID Count report data from the database. */
     payIdCountRefreshIntervalInSeconds:
       Number(process.env.PAYID_COUNT_REFRESH_INTERVAL) || 60,
+
+    payIdProtocolVersion: payIdServerVersions[payIdServerVersions.length - 1],
+
+    serverAgent: `@payid-org/payid:${
+      process.env.npm_package_version ?? 'unknown.version'
```
Comment (dino-rodriguez) [inline src/config.ts:66]: Is this variable always in the environment when a node app is started?

Comment (hbergren) [reply src/config.ts:66]: Echoing Dino, I assume you've tested this, right?

**Comment by dino-rodriguez** [comment]
Whoops, I just accidentally re-requested @hbergren for a review on this PR, meant to do it on the original metrics PR.

File: src/config.ts (hunk 2 above)
Comment (nhartner) [reply src/config.ts:66]: Yeah, good question. I only tested it using `npm start` which works, but it doesn't work inside the docker container when started using `node ...`.

Comment (nhartner) [reply src/config.ts]: Pushed a change to read this from package.json instead which I verified works using both npm and node to run the server.

File: package.json [hunk 3]
```diff
@@ -16,12 +16,13 @@
...
     "build": "run-script-os",
-    "build:default": "rm -rf build && tsc --project . && npm run copySQL && npm run copyHTML && npm run copyJSON",
-    "build:win32": "del /q /f build && tsc --project . && npm run copySQL && npm run copyHTML && npm run copyJSON",
+    "build:default": "rm -rf build && tsc --project . && npm run copySQL && npm run copyHTML && npm run copyJSON && npm run copyPackageJSON",
+    "build:win32": "del /q /f build && tsc --project . && npm run copySQL && npm run copyHTML && npm run copyJSON && npm run copyPackageJSON",
     "buildWatch": "tsc --watch --project .",
     "copyHTML": "copyfiles -u 1 src/html/* build/src",
     "copySQL": "copyfiles -u 1 **/*.sql build/src",
     "copyJSON": "copyfiles -u 1 src/**/*.json build/src",
+    "copyPackageJSON": "copyfiles package.json build",
```
Comment (hbergren) [inline package.json:25]: Really clever solution.

**Pull request merged**